module/executiondatasync/optimistic_sync/execution_result/info_provider.go

```diff
@@ -62,6 +66,11 @@ func (e *Provider) ExecutionResultInfo(
 		return nil, fmt.Errorf("failed to retrieve execution IDs: %w", err)
 	}
 
+	err = e.validateRequiredExecutors(criteria.RequiredExecutors, executorIdentities)
```

Comment:
since we're validating that the executors exist in the identity table, I think we need to use `state.AtBlockID()` to get the executors  for the specific block queried. ENs generally don't change often, but this is more correct.